x-pack/metricbeat/module/aws/elb/elb_integration_test.go

```diff
@@ -7,18 +7,55 @@
 package elb
 
 import (
+	"fmt"
 	"testing"
 
+	"github.com/elastic/beats/libbeat/common"
 	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
 	"github.com/elastic/beats/x-pack/metricbeat/module/aws/mtest"
 )
 
 func TestData(t *testing.T) {
+	namespaceIs := func(namespace string) func(e common.MapStr) bool {
+		return func(e common.MapStr) bool {
+			v, err := e.GetValue("aws.cloudwatch.namespace")
+			return err == nil && v == namespace
+		}
+	}
+
+	dataFiles := []struct {
+		namespace string
+		path      string
+	}{
+		{"AWS/ELB", "./_meta/data.json"},
+		{"AWS/ApplicationELB", "./_meta/data_alb.json"},
+		{"AWS/NetworkELB", "./_meta/data_nlb.json"},
+	}
+
 	config, info := mtest.GetConfigForTest("elb", "300s")
 	if info != "" {
 		t.Skip("Skipping TestData: " + info)
 	}
 
-	metricSet := mbtest.NewFetcher(t, config)
-	metricSet.WriteEvents(t, "/")
+	for _, df := range dataFiles {
+		metricSet := mbtest.NewFetcher(t, config)
+
+		event, errs := metricSet.FetchEvents()
```

Comment:
We shouldn't need to call to `FetchEvents()` in `TestData`, only to `WriteEvents`, that should already check if some event is returned, and skip if no data should be generated. Was this not working?

Reply:
It's working now. Somehow I missed it when first time trying it 😬 Thanks for pointing it out!